File: cs214/projects/proj08/clojure/src/temp_driver.clj
;; temp_driver.clj tests a Clojure Temperature type,
;;  stored in temperature.clj (in the src directory).
;;
;; Output: the results of testing the Temperature functions.
;;
;; Usage: clojure -m temp_driver
;;
;; Begun by: Prof. Adams, CS 214 at Calvin College.
;; Completed by: Advait Scaria
;; Date: 04/11/2019
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(ns temp_driver)

;;(load "Name")
(load-file "src/temperature.clj")

;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;; A simple driver to test Temperature functions. 
;; Output: the results of testing Temperature functions.
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
(defn -main []
  (let
    [
    		baseTemp (readTemp)
    		limitTemp (readTemp)
    ]
    (println "Enter stepValue: ")
    (let [ stepValue (read) ]
    		(newline) (println "      Fahrenheit          Celsius         Kelvin") (newline)
		    (loop [i stepValue]
		    		(when (or (lessThan (raise baseTemp i) limitTemp)
		    	          (equals (raise baseTemp i) limitTemp))
		    				(print "     ")
                (displayTemp (getInF (raise baseTemp i)))
                (print "     ")
                (displayTemp (getInC (raise baseTemp i)))
                (print "     ")
                (displayTemp (getInK (raise baseTemp i)))
                (println)
 		    			  (recur (+ i stepValue))
		    		);closes when 
		    ); closes loop
		  ) 
  )
)

File: cs214/projects/proj08/ruby/temp_driver.rb
# temp_driver.rb tests class Temperature and its operations
#
# Begun by: Dr. Adams, for CS 214 at Calvin College.
# Completed by: Advait Scaria
# Date: 04/11/2019
####################################################

require 'test/unit/assertions'
include Test::Unit::Assertions

require './temperature.rb'
include Temperatures

def temp_driver
  
  baseTemp = Temperature.new(0.0, 'K')
  limitTemp = Temperature.new(0.0, 'K')

  print "baseTemp "
  baseTemp = baseTemp.readTemp()

  print "limitTemp "
  limitTemp = limitTemp.readTemp()

  print "stepValue: "
  stepValue = gets.chomp.to_f 

  puts "\n"
  puts "\t Fahrenheit\t\t Celsius\t\tKelvin"
  puts "\n"
  # tempStepValue = 0.0
  while (baseTemp.lessThan(limitTemp) or baseTemp.equals(limitTemp) )
    baseTemp.getInF().displayTemp()
    baseTemp.getInC().displayTemp()
    baseTemp.getInK().displayTemp()
    puts "\n"
    baseTemp = baseTemp.raiseTemp(stepValue)
  end
  puts "\n"
end

temp_driver

